File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/match/TeamStatDao.java
package com.legue.axel.lolsummonertool.database.dao.match;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.match.TeamStat;

import java.util.List;
@Dao
public interface TeamStatDao {

    @Query("SELECT * FROM teamStats ORDER BY id")
    LiveData<List<TeamStat>> getTeamStats();

    @Query("SELECT * FROM teamStats WHERE matchId = :matchId")
    LiveData<TeamStat> getTeamStat(long matchId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllTeamStats(List<TeamStat> teamStats);

    @Query("DELETE FROM teamStats")
    void deleteAll();

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateTeamStat(TeamStat teamStat);
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/match/TeamStatDao.kt
package com.legue.axel.lolsummonertool.database.dao.match

import androidx.lifecycle.LiveData
import androidx.room.*
import com.legue.axel.lolsummonertool.database.model.match.TeamStat

@Dao
interface TeamStatDao {

    @Query("SELECT * FROM teamStats ORDER BY id")
    fun getTamStats(): LiveData<List<TeamStat>>

    @Query("SELECT * FROM teamStats WHERE matchId = :matchId")
    fun getTeamStat(matchId: Long): LiveData<TeamStat>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllTeamStats(teamStats: List<TeamStat>)

    @Query("DELETE FROM teamStats")
    fun deleteAll()

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateTeamStat(teamStat: TeamStat)
}
